tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc

```diff
@@ -366,18 +368,31 @@ Status CreateTRTNode(const ConversionParams& params,
     } else {
       // Data edges
       if (!conn.is_input_edge) {
-        // Set the data types of output edge.
+        // Set the shapes and data types of the output edge.
+        tensorflow::TensorShapeProto out_shape;
+        conn.inside_shape.AsProto(&out_shape);
+
+        if (output_shapes.size() <= conn.port_number) {
+          output_shape_protos.resize(conn.port_number + 1);
+          output_shapes.resize(conn.port_number + 1);
+        }
+        output_shape_protos.at(conn.port_number) = out_shape;
+        output_shapes.at(conn.port_number) = conn.inside_shape;
```

Comment:
Can we find a way to share this code snippet with the input case below?

Reply:
I'm gonna try to do smthg on that. WIP

Reply:
Fixed